optimizely/config_manager.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2019, Optimizely
```

Comment:
Can we split this into different files? Ideally `BaseConfigManager` is it's own so that users looking to write their own impl can just import that and not have to import all these other implementations that we offer

Reply:
Yeah, ideally. I had been going back and forth on how to organize this. Will think more about this.

@travisbeck @nickdirienzo @onufryk any thoughts?

Reply:
I'm personally okay if they're in the same module, as the implementations seem pretty standard and  the module is fairly small (< 200 lines). I'm also okay if you want to split the base class out, but I don't find it necessary. Just my 2c. :)